### PR TITLE
Write events when autoprovisioned nodegroup is created / deleted

### PR DESCRIPTION
**Pull request by MaciekPytel**

### CODE REVIEW

**Comment by MaciekPytel** [comment]
```
6m          6m           1         cluster-autoscaler-status.14f0cfd8be4af590                                        ConfigMap                                                Normal    CreatedNodeGroup        cl
uster-autoscaler                                             NodeAutoprovisioning: create new node group [groupname]
6m          6m           1         cluster-autoscaler-status.14f0cfd8ed1ef100                                        ConfigMap                                                Normal    ScaledUpGroup           cl
uster-autoscaler                                             Scale-up: group [groupname] size set to 1
4m          4m           1         cluster-autoscaler-status.14f0cffae5198a13                                        ConfigMap                                                Normal    ScaleDownEmpty          cl
uster-autoscaler                                             Scale-down: removing empty node [nodename]
3m          3m           1         cluster-autoscaler-status.14f0d009014a3a30                                        ConfigMap                                                Normal    DeletedNodeGroup        cl
uster-autoscaler                                             NodeAutoprovisioning: removed node group [groupname]
```

File: cluster-autoscaler/core/scale_up.go [hunk 1]
```diff
@@ -224,6 +224,9 @@ func ScaleUp(context *AutoscalingContext, unschedulablePods []*apiv1.Pod, nodes
 					nodeInfos[newId] = nodeInfos[oldId]
 					delete(nodeInfos, oldId)
 				}
+				context.LogRecorder.Eventf(apiv1.EventTypeNormal, "CreatedNodeGroup",
+					"NodeAutoprovisioning: create new node group %v", newId)
```
Comment (bskiba) [inline cluster-autoscaler/core/scale_up.go]: I'd keep time consistent, you have both deleted and create, should it be past or present?

File: cluster-autoscaler/core/scale_down.go [hunk 2]
```diff
@@ -832,9 +833,12 @@ func cleanUpNodeAutoprovisionedGroups(cloudProvider cloudprovider.CloudProvider)
 			return err
 		}
 		if size == 0 {
+			ngId := nodeGroup.Id()
 			if err := nodeGroup.Delete(); err != nil {
```
Comment (bskiba) [inline cluster-autoscaler/core/scale_down.go]: Do we want to record failure too?

File: cluster-autoscaler/core/scale_up.go (hunk 1 above)
Comment (aleksandra-malinowska) [reply cluster-autoscaler/core/scale_up.go]: +1, also "NodeGroupCreated" would be more consistent with other events in Kubernetes (like "PodScheduled")

File: cluster-autoscaler/core/scale_down.go [hunk 3]
```diff
@@ -832,9 +833,12 @@ func cleanUpNodeAutoprovisionedGroups(cloudProvider cloudprovider.CloudProvider)
...
 				return err
 			}
+			logRecorder.Eventf(apiv1.EventTypeNormal, "DeletedNodeGroup",
```
Comment (aleksandra-malinowska) [inline cluster-autoscaler/core/scale_down.go]: nit: I'd go with "NodeGroupDeleted" (see below comment). Feel free to ignore it though

File: cluster-autoscaler/core/scale_up.go (hunk 1 above)
Comment (MaciekPytel) [reply cluster-autoscaler/core/scale_up.go]: fixed tense in message to be consistent

File: cluster-autoscaler/core/scale_down.go (hunk 2 above)
Comment (MaciekPytel) [reply cluster-autoscaler/core/scale_down.go]: Now we do :)

**Pull request merged**